jupyter_fsspec/handlers.py

```diff
@@ -251,6 +251,77 @@ def post(self):
         self.finish()
 
 
+# ====================================================================================
+# Handle Syncing Local and Remote filesystems
+# ====================================================================================
+class FilesystemSyncHandler(BaseFileSystemHandler):
+    def initialize(self, fs_manager):
+        self.fs_manager = fs_manager
+
+    async def get(self):
+        # remote to local (fetching latest changes)
+        request_data = json.loads(self.request.body.decode("utf-8"))
+        local_destination_path = request_data.get("local_path")
+        remote_source_path = request_data.get("remote_path")
+        dest_fs_key = request_data.get("destination_key")
+        dest_fs_info = self.fs_manager.get_filesystem(dest_fs_key)
+        dest_path = dest_fs_info["path"]
+
+        response = {"content": None}
+
+        fs, dest_path = self.validate_fs("post", dest_fs_key, dest_path)
+        remote_fs_instance = fs["instance"]  # noqa: F841
+
+        try:
+            # rsync
+            # (remote_source_path, local_destination_path)
```

Comment:
So you don't trigger this yet?

Reply:
We can probably figure out here is this is going to be async; OR, we can use the wrapper to make everything async (run on a thread) to make sure we don't block.